src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_params.py

```diff
@@ -103,12 +103,19 @@ def get_hostname_completion_list(prefix, action, parsed_args, **kwargs):  # pyli
     register_cli_argument(scope + ' deployment source', 'manual_integration', action='store_true', help='disable automatic sync between source control and web')
     register_cli_argument(scope + ' deployment source', 'repo_url', options_list=('--repo-url', '-u'), help='repository url to pull the latest source from, e.g. https://github.com/foo/foo-web')
     register_cli_argument(scope + ' deployment source', 'branch', help='the branch name of the repository')
-    register_cli_argument(scope + ' deployment source', 'cd_provider', help='type of CI/CD provider', default='kudu', **enum_choice_list(['kudu', 'vsts']))
-    register_cli_argument(scope + ' deployment source', 'cd_app_type', arg_group='VSTS CD Provider', help='web application framework you used to develop your app', default='AspNetWap', **enum_choice_list(['AspNetWap', 'AspNetCore', 'NodeJSWithGulp', 'NodeJSWithGrunt']))
-    register_cli_argument(scope + ' deployment source', 'cd_account', arg_group='VSTS CD Provider', help='name of the Team Services account to create/use for continuous delivery')
-    register_cli_argument(scope + ' deployment source', 'cd_account_must_exist', arg_group='VSTS CD Provider', help='specifies that the account must already exist. If not specified, the account will be created if it does not already exist (existing accounts are updated)', action='store_true')
+    register_cli_argument(scope + ' deployment source', 'private_repo_username', arg_group='VSTS CD Provider', help='Username for the private repository')
+    register_cli_argument(scope + ' deployment source', 'private_repo_password', arg_group='VSTS CD Provider', help='Password for private repository. Will prompt if not provided')
+    register_cli_argument(scope + ' deployment source', 'cd_app_type', arg_group='VSTS CD Provider', help='Web application framework you used to develop your app', default='AspNet', **enum_choice_list(['AspNet', 'AspNetCore', 'NodeJS', 'PHP', 'Python']))
+    register_cli_argument(scope + ' deployment source', 'app_working_dir', arg_group='VSTS CD Provider', help='Working directory of the application. Default will be root of the repo')
+    register_cli_argument(scope + ' deployment source', 'nodejs_task_runner', arg_group='VSTS CD Provider', help='Task runner for nodejs. Default : None', default='None', **enum_choice_list(['None', 'Gulp', 'Grunt']))
+    register_cli_argument(scope + ' deployment source', 'python_framework', arg_group='VSTS CD Provider', help='Framework used for Python application. Default : Django', default='Django', **enum_choice_list(['Bottle', 'Django', 'Flask']))
```

Comment:
Instead of putting the defaults here, I would recommend you put them as the default values in your command signature. It is more self-documenting, and the CLI will then automatically pick them up as the default and recognize that they are optional.